### PR TITLE
Search improved

### PR DESCRIPTION
**Pull request by DorotaKar**
- add NoResultsPage for movie search with no results
- add NoResultsPage for empty search results
-  set dynamic section title based on search state

### CODE REVIEW

**Review by maxFadet — APPROVED** [review]

File: src/features/MoviesListPage/MainContent/index.js [hunk 1]
```diff
@@ -38,13 +39,17 @@ export const MainContent = () => {
 
     const moviesToDisplay = isSearching ? searchResults : popularMovies.results;
 
+    if (isSearching && (!moviesToDisplay || moviesToDisplay.length === 0)) {
+        return <NoResults query={searchQuery} />;
```
Comment (OskarWlaszczuk) [inline src/features/MoviesListPage/MainContent/index.js:43]: Good job, but I have a question!
Is it necessary to write both of this requirements in statemant?
**(!moviesToDisplay || moviesToDisplay.length === 0))**
Maybe just !moviesToDisplay or moviesToDisplay.length === 0) will be enough?

**Review by OskarWlaszczuk — CHANGES_REQUESTED** [review]
Great improves

**Review by OskarWlaszczuk — APPROVED** [review]

File: src/features/MoviesListPage/MainContent/index.js (hunk 1 above)
Comment (DorotaKar) [reply src/features/MoviesListPage/MainContent/index.js:43]: You're right, using just !moviesToDisplay or moviesToDisplay.length === 0 is a cleaner way to check if the array is empty.
I'll update the code to use your suggested condition.

Comment (DorotaKar) [reply src/features/MoviesListPage/MainContent/index.js]: The condition is necessary to display the <NoResults> component with the image. Without this check, the <NoResults> component would render correctly, but the image would not be shown, providing an incomplete user experience.

**Pull request merged**